internal/tofu/transform_targets.go

```diff
@@ -13,34 +13,42 @@ import (
 )
 
 // GraphNodeTargetable is an interface for graph nodes to implement when they
-// need to be told about incoming targets. This is useful for nodes that need
-// to respect targets as they dynamically expand. Note that the list of targets
-// provided will contain every target provided, and each implementing graph
-// node must filter this list to targets considered relevant.
+// need to be told about incoming targets or excluded targets. This is useful for
+// nodes that need to respect targets and excludes as they dynamically expand.
+// Note that the lists of targets and excludes provided will contain every target
+// or every exclude provided, and each implementing graph node must filter this
+// list to targets considered relevant.
 type GraphNodeTargetable interface {
 	SetTargets([]addrs.Targetable)
+	SetExcludes([]addrs.Targetable)
 }
 
 // TargetsTransformer is a GraphTransformer that, when the user specifies a
-// list of resources to target, limits the graph to only those resources and
-// their dependencies.
+// list of resources to target, or a list of resources to exclude, limits the
+// graph to only those resources and their dependencies (or in the case of
+// excludes - limits the graph to all resources that are not excluded or not
+// dependent on excluded resources).
 type TargetsTransformer struct {
```

Comment:
Can I ask why you decided to add it to this existing transformer instead of keeping it separate? For navigation of the code I would expect to find a `ExcludesTransformer` somewhere and this may be confusing

Reply:
We'd probably prefer this being a single transformer, as right now "targeting" is a concept that both `-target` and `-exclude` flag contribute to. And, if in the future we'd want to support both `-target` and -`exclude` flag together, it would make more sense for the transformer to be singular.

I've renamed it to TargetingTransformer - 07779eb70d1c7e048f35c3eb8e1bf5c88f785df4